File: src/main/java/frc/robot/subsystems/climber/ClimberIOSim.java
package frc.robot.subsystems.climber;

import edu.wpi.first.math.system.plant.DCMotor;
import edu.wpi.first.wpilibj.simulation.DCMotorSim;
import frc.robot.constants.ClimberConstants;

public class ClimberIOSim extends ClimberIO {
    private final DCMotorSim motor = new DCMotorSim(DCMotor.getNEO(1), ClimberConstants.gearRatio, 0.0001);
    private double appliedVolts = 0.0;

    @Override
    public void updateInputs(ClimberIOInputs inputs) {
        motor.update(0.02);

        inputs.positionRad = motor.getAngularPositionRad();
        inputs.appliedVolts = appliedVolts;
        inputs.currentAmps = motor.getCurrentDrawAmps();
    }

    @Override
    public void set(double volts) {
        appliedVolts = volts;
        motor.setInputVoltage(volts);
    }

    @Override
    public void stop() {
        appliedVolts = 0;
        motor.setInputVoltage(0);
    }

    @Override
    public void resetPosition() {
        motor.setState(0, 0);
    }
}

File: src/main/java/frc/robot/subsystems/climber/ClimberIOSim.kt
package frc.robot.subsystems.climber

import edu.wpi.first.math.system.plant.DCMotor
import edu.wpi.first.wpilibj.simulation.DCMotorSim
import frc.robot.Constants

class ClimberIOSim : ClimberIO() {
    private val motor = DCMotorSim(DCMotor.getNEO(1), Constants.Climber.gearRatio, 0.0001)
    private var appliedVolts = 0.0

    override fun updateInputs(inputs: ClimberIOInputs) {
        motor.update(0.02)

        inputs.positionRad = motor.angularPositionRad
        inputs.appliedVolts = appliedVolts
        inputs.currentAmps = motor.currentDrawAmps
    }

    override fun set(volts: Double) {
        appliedVolts = volts
        motor.setInputVoltage(volts)
    }

    override fun stop() {
        appliedVolts = 0.0
        motor.setInputVoltage(0.0)
    }

    override fun resetPosition() {
        motor.setState(0.0, 0.0)
    }
}
